### PR TITLE
Disable p2p network when delayed_node plugin is enabled

### LINKED ISSUE #104: delayed_node connects to the P2P network
_From @xeroc on October 22, 2015 7:43_

People need to run this

```
./programs/delayed_node/delayed_node --trusted-node="127.0.0.1:8090" -d delayed_node -s "0.0.0.0:0" --p2p-endpoint="0.0.0.0:0" --seed-nodes "[]"
```

in order for the delayed node not to connect to the internet via P2P .. the default seed nodes are in place otherwise

_Copied from original issue: cryptonomex/graphene#388_

### PR DESCRIPTION
**Pull request by pmconrad**
Resolves #104

### CODE REVIEW

**Comment by abitmore** [comment]
Better cover `network_broadcast_api` as well.

**Comment by pmconrad** [comment]
`network_broadcast_api` methods already contain null-checks for `p2p_node()`. Question is if an error reply would be more appropriate than silently dropping the stuff to be broadcasted.
Also, `broadcast_transaction_with_callback` and `broadcast_transaction_synchronous` leak memory in this case because the callback is never executed. (The memory still gets free'd when the client closes the connection.)

**Comment by abitmore** [comment]
> Question is if an error reply would be more appropriate than silently dropping the stuff to be broadcasted.

I think raising an error is better.

File: libraries/app/api.cpp [hunk 1]
```diff
@@ -180,9 +180,9 @@ namespace graphene { namespace app {
     void network_broadcast_api::broadcast_transaction(const precomputable_transaction& trx)
     {
        _app.chain_database()->precompute_parallel( trx ).wait();
+       FC_ASSERT( _app.p2p_node() != nullptr, "Not connected to P2P network, can't broadcast!" );
        _app.chain_database()->push_transaction(trx);
-       if( _app.p2p_node() != nullptr )
-          _app.p2p_node()->broadcast_transaction(trx);
+       _app.p2p_node()->broadcast_transaction(trx);
```
Comment (abitmore) [inline libraries/app/api.cpp]: Hmm, I guess this change would break at least a test case.

**Comment by pmconrad** [comment]
Did some refactoring in `database_fixture`. This could also be a first step towards #1907 .

**Pull request merged**